compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -902,12 +904,15 @@ impl<'a> Parser<'a> {
                 };
                 let symbol1 = Symbol::intern(&i1);
                 self.token = Token::new(token::Ident(symbol1, false), ident1_span);
-                let next_token1 = Token::new(token::Dot, dot_span);
+                // This `.` is part of a floating point literal, so it cannot
+                // be glued to anything else (e.g. to form a  `..`)
+                let next_token1 = (Token::new(token::Dot, dot_span), Spacing::Alone);
```

Comment:
This is the dot in the middle of `1.2e3`, so it's definitely joint, but it is followed by an identifier and https://github.com/rust-lang/rust/pull/76285 was reverted due to some regression, so it has to be `Spacing::Alone`, but not due to the reason mentioned in the comment.